recipe/cross_compile_support.sh

```diff
@@ -15,19 +22,79 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
     fi
     if [[ "${BUILD_PLATFORM}" == "linux-64" && "${HOST_PLATFORM}" == linux-* ]]; then
         mamba create -n sysroot_${HOST_PLATFORM} --yes --quiet sysroot_${HOST_PLATFORM}
+        HOST_PLATFORM_ARCH=${HOST_PLATFORM:6}
         if [[ -f ${CI_SUPPORT}/../recipe/yum_requirements.txt ]]; then
             for pkg in $(cat ${CI_SUPPORT}/../recipe/yum_requirements.txt); do
                 if [[ "${pkg}" != "#"* && "${pkg}" != "" ]]; then
-                    mamba install "${pkg}-cos7-${HOST_PLATFORM:6}" -n sysroot_${HOST_PLATFORM} --yes --quiet || true
+                    mamba install "${pkg}-cos7-${HOST_PLATFORM_ARCH}" -n sysroot_${HOST_PLATFORM} --yes --quiet || true
                 fi
             done
         fi
         export QEMU_LD_PREFIX=$(find ${CONDA_PREFIX}/envs/sysroot_${HOST_PLATFORM} -name sysroot | head -1)
         if [ -f ${CI_SUPPORT}/${CONFIG}.yaml ]; then
             echo "CMAKE_CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
             echo "CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+        fi
+
+
+        if [[ "${CUDA_COMPILER_VERSION}" == "11.2" ]] && [[ "${CDT_NAME}" == "cos8" || "${CDT_NAME}" == "cos7" ]]; then
+            # We use cdt_name=cos7 for rhel8 based nvcc till we figure out
+            # a stable cos8 replacement.
+            EXTRACT_DIR=$(mktemp -d)
+            pushd ${EXTRACT_DIR}
+                if [[ "${HOST_PLATFORM_ARCH}" == "aarch64" ]]; then
+                    CUDA_HOST_PLATFORM_ARCH=sbsa
+                else
+                    CUDA_HOST_PLATFORM_ARCH=${HOST_PLATFORM_ARCH}
+                fi
+                # download manifest for latest 11.2.x patch version
+                curl -L https://developer.download.nvidia.com/compute/cuda/repos/rhel8/${CUDA_HOST_PLATFORM_ARCH}/version_11.2.2.json > versions.json
+                # packages for which we also need to install the -devel version
+                # (names need "_" not "-" to match spelling in manifest)
+                declare -a DEVELS=(
+                    "cuda_cudart"
+                    "cuda_driver"
+                    "cuda_nvml"
+                    "cuda_nvrtc"
+                    "libcublas"
+                    "libcufft"
+                    "libcurand"
+                    "libcusolver"
+                    "libcusparse"
+                    "libnpp"
+                    "libnvjpeg"
+                )
```

Comment:
Need to add: `nvidia_driver`, `nvidia_fabric_manager`, `nvidia_libXNVCtrl`. All three should use the version from the `nvidia_driver` key

Reply:
OK, this and the `cuda_driver` thing make it sound like we need to do a full mapping here. Currently it was just `<x> -> `<x>_devel`.


You're saying I need to change this to:
```
    # new key...                ...uses version from
    cuda_driver_devel       ->  cuda_cudart       # special
    cuda_nvrtc_devel        ->  cuda_nvrtc
    cuda_nvml_devel         ->  cuda_nvml_dev     # special
    libcublas_devel         ->  libcublas
    libcufft_devel          ->  libcufft
    libcurand_devel         ->  libcurand
    libcusolver_devel       ->  libcusolver
    libcusparse_devel       ->  libcusparse
    libnpp_devel            ->  libnpp
    libnvjpeg_devel         ->  libnvjpeg
    nvidia_driver_devel     ->  nvidia_driver   
    nvidia_fabric_manager   ->  nvidia_driver     # special
    nvidia_libXNVCtrl       ->  nvidia_driver     # special
```

Reply:
Those are not needed.

Reply:
Can you specify what you mean by "those"? The original list was based on the devel-libs you had in your PR, but weren't listed in the manifest

Reply:
`nvidia_libXNVCtrl` is GPL licensed (https://github.com/NVIDIA/nvidia-settings/blob/main/COPYING) and not governed under the EULA so I think (I am not a lawyer!) we can actually ship that as a conda package if desired so that can definitely be removed from the list in hindsight. My apologies.

I can't find a license or EULA for the `nvidia_fabric_manager` (should actually have been `nvidia_fabric_manager_devel`, my bad again), so I assumed we weren't allowed to ship it as a conda package. I haven't seen anyone actually use this library before so I'd be perfectly happy to leave it out.

There's other things that would potentially be needed that there aren't stub libraries available for:
- `nvidia-driver-libs` notably contains `libnvoptix` and a bunch of other libraries
- `nvidia-driver-cuda-libs` notably contains `libnvidia-nvvm` and `libnvidia-ptxjitcompiler` and more generally contains the libraries that the symlinks in `nvidia_driver_devel` point to

I have seen a fair amount of software that uses these options where I think it would make sense to add these libraries: `nvidia_driver_devel`, `nvidia_driver_libs`, and `nvidia_driver_cuda_libs`.

Notably with the proposed above, we'll be reliant on stubs for the CUDA driver library `libcuda` and the NVIDIA Management Library `libnvidia-ml`. I'm not sure how build systems like CMake will react to using a mix of actual and stub libraries. Hopefully everything will be okay?

Reply:
`nvidia-libXNVCtrl{,-devel}` and `nvidia-fabric-manager{,-devel}` don't exist under:
https://developer.download.nvidia.com/compute/cuda/repos/rhel8/cross-linux-sbsa/
https://developer.download.nvidia.com/compute/cuda/repos/rhel8/ppc64le/